libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1743,6 +1743,25 @@ void GCS_MAVLINK::handle_common_gps_message(mavlink_message_t *msg)
     AP::gps().handle_msg(msg);
 }
 
+void GCS_MAVLINK::handle_system_time_message(mavlink_message_t *msg)
+{
+    static bool system_time_set = false;
```

Comment:
I don't think this is good (besides of us trying to not have statics in methods). I think It should be checked at HAL level, I believe we don't want to ever override the system time once it's set.